swim/join_sender.go

```diff
@@ -65,6 +65,10 @@ type joinOpts struct {
 	size              int
 	maxJoinDuration   time.Duration
 	parallelismFactor int
+
+	// hostProvider is the HostProvider that this joinSender will use to
+	// enumerate bootstrap hosts.
+	hostProvider HostProvider
```

Comment:
"Host" is definitely the wrong word. We should move away from it for bootstrap-related code. I'd prefer the word Node since it somewhat abstracts away the details of a Ringpop Node mapping directly to a core or a host or a container or otherwise.